File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool comp(string &s,string &t){
        int x=s.size()+t.size();
        int i=0,j=0;
        while(x--){
            if(s[i]!=t[j])
                return s[i]>t[j];
            i++,j++;
            if(i==s.size())
                i=0;
            if(j==t.size())
                j=0;
        }
        return 0;
    }
    
    string largestNumber(vector<int>& nums) {
        vector<string>arr;
        for(int i:nums){
            arr.push_back(to_string(i));
        }
        
        sort(arr.begin(),arr.end(),comp);
        
        string s="";
        for(string t:arr)
            s+=t;
        if(s[0]=='0')
            return "0";
        
        return s;
    }
};

File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        Comparator<String> comparator = Comparator.reverseOrder();

        PriorityQueue<String>pq=new PriorityQueue<>((a,b)->{
            return comparator.compare(a+b,b+a);
        }
        );
        
        long sum=0;
        
        for(int i:nums){
            sum+=i;
            pq.add(String.valueOf(i));
        }
        if(sum==0){
            return "0";
        }
        
        StringBuilder sb=new StringBuilder();
        while(pq.size()!=0){
            sb.append(pq.poll());
        }
        
        return sb.toString();
    }
}